custom_components/switchbotremote/client/client.py

```diff
@@ -39,22 +44,43 @@ def headers(self):
 
         return headers
 
-    def request(self, method: str, path: str, **kwargs) -> Any:
+    def __request(self, method: str, path: str, **kwargs) -> Any:
         url = f"{switchbot_host}/{path}"
         _LOGGER.debug(f"Calling service {url}")
         response = request(method, url, headers=self.headers, **kwargs)
 
         if response.status_code != 200:
             _LOGGER.debug(f"Received http error {response.status_code} {response.text}")
-            raise RuntimeError(f"SwitchBot API server returns status {response.status_code}")
+            if response.status_code != 500:
+                raise HomeAssistantError(f"SwitchBot API server returns status {response.status_code}")
+            else:
+                raise SwitchbotInternal500Error
 
         response_in_json = humps.decamelize(response.json())
         if response_in_json["status_code"] != 100:
             _LOGGER.debug(f"Received error in response {response_in_json}")
-            raise RuntimeError(f'An error occurred: {response_in_json["message"]}')
+            raise HomeAssistantError(f'An error occurred: {response_in_json["message"]}')
 
         _LOGGER.debug(f"Call service {url} OK")
         return response_in_json
+    
+    def request(self, method: str, path: str, maxNumberOfTrials: int = MAX_TRIES, delayMSBetweenTrials: int = DELAY_BETWEEN_TRIES_MS, **kwargs) -> Any:
+        """Try to send the request.
+        If the server returns a 500 Internal error status, will retry until it succeeds or it passes a threshold of max number of tries.
+        Any other error will be thrown."""
+        for tryNumber in range(maxNumberOfTrials):
+            try:
+                result = self.__request(method, path, **kwargs)
+                break
```

Comment:
I have to keep the `break` statement there in order to get out of the loop if the request succeeds.
I added the `else` statement which is better than checking the `tryNumber` but I don't think that I can get rid of the `break` statement unless I completely rethink the way the trials are handled.

Reply:
@MrXANA91 As I suggested, you should use a return statement in case of success directly inside the try block, and the else statement is executed just when the loop end without result